samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -679,11 +694,45 @@ class SamzaContainer(
       shutdownMetrics
       shutdownSecurityManger
 
+      if (!status.equals(SamzaContainerStatus.FAILED)) {
+        status = SamzaContainerStatus.STOPPED
+      }
+
       info("Shutdown complete.")
+    } catch {
+      case e: Throwable =>
+        error("Caught exception/error while shutting down container.", e)
+        if (exceptionSeen == null) {
+          exceptionSeen = e
+        }
+        status = SamzaContainerStatus.FAILED
+    }
+
+    status match {
+      case SamzaContainerStatus.STOPPED =>
+        if (containerListener != null) {
+          containerListener.onContainerStop(paused)
+        }
+      case SamzaContainerStatus.FAILED =>
+        if (containerListener != null) {
+          containerListener.onContainerFailed(exceptionSeen)
```

Comment:
I actually did a usage search and that is how I identified LocalContainerRunner. I don't see any other usage of container.run. The only callers are LocalContainerRunner, ThreadJob and StreamProcessor. 
If there are custom usages of this interface through CommandBuilders (ie. using task.execute), it should be handled by those custom scripts. Let me know if you find other places of usage. I couldn't find any.